### PR TITLE
Add ConditionalReplaceValueTransformWithDefault

### PR DESCRIPTION
**Pull request by huitseeker**
A compact version of partitioning the replacement values for a column by a condition,
offshoot of work on the intrusion demo.

### CODE REVIEW

**Comment by huitseeker** [comment]
@AlexDBlack Indeed. I hope my latest fixes all that.

To be clear, this operation is strictly better than two successive conditional replaces, not just form performance, but
- in the case where we wish to have a non-empty intersection between the image of one side of the partition and the domain of the other (e.g. if x  <= 0 replace by 1 else replace by 0)
- and it makes the case where the condition is a pain to negate, such as regexes, a bit easier (than the use of Boolean condition operators).

File: datavec-api/src/main/java/org/datavec/api/transform/TransformProcess.java [hunk 1]
```diff
@@ -1327,6 +1328,19 @@ public Builder conditionalReplaceValueTransform(String column, Writable newValue
...
         /**
+         * Replace the values in a specified column with a specified new value, if some condition holds.
+         * If the condition does not hold, the original values are not modified.
+         *
+         * @param column    Column to operate on
+         * @param newVal  Value to use as replacement, if condition is satisfied
+         * @param defaultVal  Value to use as replacement, if condition is satisfied
+         * @param condition Condition that must be satisfied for replacement
+         */
+        public Builder conditionalReplaceValueTransformWithDefault(String column, Writable newVal, Writable defaultVal, Condition condition) {
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/transform/TransformProcess.java]: Should align with the ```yesVal``` and ```noVal``` semantics of the constructor.

File: datavec-api/src/main/java/org/datavec/api/transform/TransformProcess.java [hunk 2]
```diff
@@ -1327,6 +1328,19 @@ public Builder conditionalReplaceValueTransform(String column, Writable newValue
         }
 
         /**
+         * Replace the values in a specified column with a specified new value, if some condition holds.
+         * If the condition does not hold, the original values are not modified.
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/transform/TransformProcess.java]: Still incorrect, does not match class javadoc.

File: datavec-api/src/main/java/org/datavec/api/transform/TransformProcess.java [hunk 3]
```diff
@@ -1327,6 +1328,19 @@ public Builder conditionalReplaceValueTransform(String column, Writable newValue
...
+         *
+         * @param column    Column to operate on
+         * @param newVal  Value to use as replacement, if condition is satisfied
+         * @param defaultVal  Value to use as replacement, if condition is satisfied
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/transform/TransformProcess.java]: is *not* satisfied

**Comment by huitseeker** [comment]
Ouch, apologies: I messed up, I should have paid more attention to this.

**Pull request merged**